openwisp_monitoring/device/admin.py

```diff
@@ -63,10 +63,32 @@ class CheckInline(GenericStackedInline):
     fields = ['check_type', 'is_active']
     readonly_fields = ['check_type']
 
+    def _set_check_inline(self):
+        self.fields = []
+        self.readonly_fields = ['created', 'modified']
```

Comment:
I am not sure if this is a wrong way to achieve this functionality, but it does not feel right. Django provides hooks for returning `fields` and `readonly_fields`. Using them makes the code more readable.